src/ppx/grfx/vk/vk_instance.cpp

```diff
@@ -34,7 +34,7 @@ static VkBool32 VKAPI_PTR DebugUtilsMessengerCallback(
     // Ignore these messages because they're nonsense
     // clang-format off
     if (
-        (pCallbackData->messageIdNumber == 0x3d492883) // vkCreateShaderModule(): The SPIR-V Extension (SPV_GOOGLE_hlsl_functionality1) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied
+        (std::string(pCallbackData->pMessageIdName) == "VUID-VkShaderModuleCreateInfo-pCode-08742") // vkCreateShaderModule(): The SPIR-V Extension (SPV_GOOGLE_hlsl_functionality1) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied
```

Comment:
I did a quick search in the validation layer code, it is strange that we have different messageIdNumber if the pMessageIdName remains the same:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/layers/error_message/logging.cpp#L144

Unless we updated the validation layer recently so that the `hash_util::VuidHash` is different? @apazylbe

Reply:
Maybe it has to do with this + different versions of Vulkan 1.1 being used? https://www.reddit.com/r/vulkan/comments/b41ev1/messageidnumber_reported_by_vulkan_validation/

In general it just seems like the `messageIdNumber` shouldn't be used. Official documentation also suggests `messageIdName` https://docs.vulkan.org/guide/latest/validation_overview.html#_special_usage_tags

Reply:
i see, thanks for the links